### ISSUE TITLE
Stream delay

### ISSUE THREAD

**Issue by CharisardTT** [arthurbolsoni/Purple-adblock#83]
Hi,

I've installed this extension and it works really well. However, I have a lot of delay in every stream, like 10 seconds or more and I don't know why. If I turn the extension off I have less than a second so...

I use Edge if that helps. Thanks.

![asdfasfdas](https://github.com/arthurbolsoni/Purple-adblock/assets/24940126/f7364409-e446-4cbc-b6d0-da39f4791673)

**Comment by GobOTB** [comment]
Same issue in Firefox

**Comment by arthurbolsoni** [comment]
Small delay is common

**Comment by GobOTB** [comment]
for real its not small delay
also some times I see ads

**Comment by CharisardTT** [comment]
> for real its not small delay also some times I see ads

Yeah it is insane, but it works fine the most of the time. The only problem i hate is the delay, i react to things 10 seconds after normal people hahaha

**Comment by Nemisor** [comment]
More of a suggestion than an "issue" but it might help if there was a way to script setting resolution to "source" once ads are complete (similar to the youtube extension YouTube AutoHD + FPS). I've noticed several times that when they are complete, resolution doesn't revert to Auto or whatever it was previously, so if there was a way to trigger Source as part of the script that might prevent that problem.

**Comment by MarrcosabTTS** [comment]
> More of a suggestion than an "issue" but it might help if there was a way to script setting resolution to "source" once ads are complete (similar to the youtube extension YouTube AutoHD + FPS). I've noticed several times that when they are complete, resolution doesn't revert to Auto or whatever it was previously, so if there was a way to trigger Source as part of the script that might prevent that problem.

I'm using Ublock Origin with the TwitchAdSolutions script permalink (https://github.com/pixeltris/TwitchAdSolutions) and since 2 months I've never had an ad again :).
I followed this tutorial if you are interested https://youtu.be/O-DmHA4IwKo
